src/views/shared/network-box/network-box.view.tsx

```diff
@@ -1,76 +1,113 @@
-import { FC } from "react";
+import { FC, useState } from "react";
 
 import { ReactComponent as MetaMaskIcon } from "src/assets/icons/metamask.svg";
 import { useEnvContext } from "src/contexts/env.context";
 import { useProvidersContext } from "src/contexts/providers.context";
-import { Chain } from "src/domain";
+import { useUIContext } from "src/contexts/ui.context";
+import { useErrorContext } from "src/contexts/error.context";
 import Card from "src/views/shared/card/card.view";
 import useNetworkBoxStyles from "src/views/shared/network-box/network-box.styles";
 import Typography from "src/views/shared/typography/typography.view";
+import { isMetaMaskUserRejectedRequestError } from "src/utils/types";
+import useCallIfMounted from "src/hooks/use-call-if-mounted";
+import { parseError } from "src/adapters/error";
+import { Message } from "src/domain";
 
-interface NetworkBoxProps {
-  isAddNetworkButtonDisabled: boolean;
-  onChangeNetwork: (chain: Chain) => void;
-}
-
-const NetworkBox: FC<NetworkBoxProps> = ({ isAddNetworkButtonDisabled, onChangeNetwork }) => {
+const NetworkBox: FC = () => {
   const classes = useNetworkBoxStyles();
   const env = useEnvContext();
-  const { connectedProvider } = useProvidersContext();
+  const { connectedProvider, addNetwork } = useProvidersContext();
+  const [isAddNetworkButtonDisabled, setIsAddNetworkButtonDisabled] = useState(false);
+  const { openSnackbar } = useUIContext();
+  const callIfMounted = useCallIfMounted();
+  const { notifyError } = useErrorContext();
 
   if (!env) {
     return null;
   }
 
+  const ethereumChain = env.chains[0];
+  const polygonZkEVMChain = env.chains[1];
+
+  const successMsg: Message = {
+    type: "success-msg",
+    text: `${polygonZkEVMChain.name} network successfully added`,
+  };
+
+  const onAddNetwork = (): void => {
+    setIsAddNetworkButtonDisabled(true);
+    addNetwork(polygonZkEVMChain)
+      .then(() => {
+        setIsAddNetworkButtonDisabled(false);
+        callIfMounted(() => {
+          openSnackbar(successMsg);
+        });
+      })
+      .catch((error) => {
+        callIfMounted(() => {
+          setIsAddNetworkButtonDisabled(false);
```

Comment:
I would move this line to a `finally` to avoid repeating it.

Reply:
Yep, done.